includes/html/api_functions.inc.php

```diff
@@ -2249,6 +2249,44 @@ function add_service_for_host(\Illuminate\Http\Request $request)
     return api_error(500, 'Failed to add the service');
 }
 
+function add_parents_to_host(\Illuminate\Http\Request $request)
+{
+    $data = json_decode($request->getContent(), true);
+    $device_id = $request->route('id');
+    $parent_ids = explode(',', $data['parent_ids']);
+    if (validateDeviceIds($parent_ids) && validateDeviceIds(array($device_id)) && (!in_array($device_id, $parent_ids))) {
+        Device::find($device_id)->parents()->sync($parent_ids);
+        return api_success_noresult(201, 'Device dependencies have been saved');
+    }
+    return api_error(400, "Check your parent and device IDs");
+}
+
+function del_parents_from_host(\Illuminate\Http\Request $request)
+{
+    $device_id = $request->route('id');
+    $data = json_decode($request->getContent(), true);
+    if (!validateDeviceIds(array($device_id))) {
+        return api_error(400, "Check your device ID!");
+    }
+    $device = Device::find($device_id);
+    $result = (empty($data['parent_ids']) ? $device->parents()->detach():(validateDeviceIds(explode(',', $data['parent_ids'])) ? $device->parents()->detach(explode(',', $data['parent_ids'])):false));
```

Comment:
nice oneliner,
but could you please change it to a regular if statement, so it's easier to read.
Also line length breaks [Code Guidelines](https://docs.librenms.org/Developing/Code-Guidelines/)
thx

Reply:
hmmm, let me see what i can do, our dear friend codeclimate doesn't like my if-else combos XD.

Reply:
code climate is a guideline, but not essential for merging

Reply:
hahahaha i didn't know that, let's see if find a way to keep codeclimate happy and comply with line lenght/code understandability, if not will go for the second one.

Reply:
Made another change looks more readable. Tried to change the code to keep codeclimate happy too. If this doesn't work for the project i will have to put something like this:

`if (empty($data['parent_ids'])) {
        //no parents ids has been set in the request
        $result = $device->parents()->detach(); //remove all parents
    } else {
        $parents = explode(',', $data['parent_ids']);
        $result = validateDeviceIds($parents)?$device->parents()->detach($parents):false;
    }`

I know code climate won't like this ;)